### PR TITLE
Updated Dropdown for Clearable Selection

### PR DESCRIPTION
**Pull request by saidhruv**

### CODE REVIEW

File: packages/base/components/Input/Dropdown.js [hunk 1]
```diff
@@ -110,11 +115,7 @@ const Dropdown = ({ input, ...rest }) => {
 };
 
 Dropdown.propTypes = {
-  input: PropTypes.shape({
-    name: PropTypes.string.isRequired,
-    onChange: PropTypes.func.isRequired,
```
Comment (cadejscroggins) [inline packages/base/components/Input/Dropdown.js]: Confused why you removed these.

File: packages/base/components/Input/Dropdown.js [hunk 2]
```diff
@@ -58,20 +55,28 @@ const Dropdown = ({ input, ...rest }) => {
...
+                const parsedValue = Array.isArray(value)
+                  ? value.map(o => o.value)
+                  : value.value;
 
-              if (inputRest.onChange) inputRest.onChange(parsedValue, meta);
-              input.onChange(parsedValue, meta);
+                if (inputRest.onChange) inputRest.onChange(parsedValue, meta);
+                input.onChange(parsedValue, meta);
+              } else {
+                input.onChange('', meta);
```
Comment (cadejscroggins) [inline packages/base/components/Input/Dropdown.js]: I think we'd still want to call `inputRest.onChange` in this case. Let's only set `parsedValue` within the `if (value)` condition and keep the `onChange` calls separate.

File: packages/base/components/Input/Dropdown.js [hunk 3]
```diff
@@ -58,20 +55,28 @@ const Dropdown = ({ input, ...rest }) => {
...
 
-            const value = Array.isArray(input.value)
-              ? input.value.map(optionByValue)
-              : optionByValue(input.value);
+            setValue(
+              input.value !== ''
+                ? Array.isArray(input.value)
+                  ? input.value.map(optionByValue)
+                  : optionByValue(input.value)
+                : ''
```
Comment (cadejscroggins) [inline packages/base/components/Input/Dropdown.js]: Could we have the same effect by returning an empty string from `optionByValue` if the option isn't found (by appending `|| ''` to that function)? Additionally this nested ternary operation is most likely why the build is failing.

File: packages/base/components/Input/Dropdown.js [hunk 4]
```diff
@@ -58,20 +55,28 @@ const Dropdown = ({ input, ...rest }) => {
...
+              }
             };
 
             const optionByValue = value =>
               (inputRest.options || asyncOptions).find(o => o.value === value);
 
-            const value = Array.isArray(input.value)
-              ? input.value.map(optionByValue)
-              : optionByValue(input.value);
+            setValue(
```
Comment (cadejscroggins) [inline packages/base/components/Input/Dropdown.js]: Just realized you are setting this value on every render.. Why was the component managed `value` state added?

File: packages/base/components/Input/Dropdown.js (hunk 1 above)
Comment (saidhruv) [reply packages/base/components/Input/Dropdown.js]: This is unintentional. Had the older version.

Comment (saidhruv) [reply packages/base/components/Input/Dropdown.js]: This is unintentional. Had an older version.

File: packages/base/components/Input/Dropdown.js (hunk 4 above)
Comment (saidhruv) [reply packages/base/components/Input/Dropdown.js]: I was trying to emulate a solution. Didn't realize I had a simpler option.

**Pull request merged**